cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/SparkTwillRunnable.java

```diff
@@ -49,12 +47,15 @@ public SparkTwillRunnable(String name) {
 
   @Override
   protected ProgramRunner createProgramRunner(Injector injector) {
-    // Inside the TwillRunanble, we use the "Local" SparkRunner, since we need to actually submit the job.
+    // Inside the TwillRunnable, we use the "Local" SparkRunner, since we need to actually submit the job.
     // The actual execution mode of the job is governed by the framework configuration,
     // which is in the hConf we shipped from DistributedSparkProgramRunner
-    CConfiguration cConf = injector.getInstance(CConfiguration.class);
-    return new SparkProgramRuntimeProvider(SparkCompatReader.get(cConf)) { }
-      .createProgramRunner(ProgramType.SPARK, ProgramRuntimeProvider.Mode.LOCAL, injector);
+    ProgramRuntimeProviderLoader runtimeProviderLoader = injector.getInstance(ProgramRuntimeProviderLoader.class);
```

Comment:
if we do it this way, is there other code that can now be removed? Would imagine there is some other layer that used to be doing this before?